src/python/pants/backend/jvm/targets/jvm_target.py

```diff
@@ -137,23 +137,33 @@ def strict_deps(self):
 
   @property
   def exports(self):
+    return self.payload.exports
+
+  @property
+  def exports_targets(self):
     """A list of exported targets, which will be accessible to dependents.
 
     :return: See constructor.
     :rtype: list
     """
-    exports = []
+    exports_targets = []
     for spec in self.payload.exports:
       addr = Address.parse(spec, relative_to=self.address.spec_path)
       target = self._build_graph.get_target(addr)
+      if target.is_thrift:
```

Comment:
It feels like this ought to be more general. In theory, it could apply to any gen target that's exported. I'm not sure whether that makes sense to do in this review though.

I'd also like to see some unit tests for this method that exercise the new functionality.